### PR TITLE
VR > Image Proportions > Wild-west

### PR DESCRIPTION
**Pull request by juanfra**
🎫 https://central.tri.be/issues/135820
🎥 https://www.loom.com/share/a51273a352ae4196b18cf675bd6955ec

### CODE REVIEW

File: src/views/v2/day/event/featured-image.php [hunk 1]
```diff
@@ -28,14 +28,18 @@
...
-			<div
-				class="tribe-common-c-image__bg"
-				style="background-image: url('<?php echo esc_url( $event->thumbnail->full->url ); ?>');"
-				role="img"
-				aria-label="<?php echo esc_attr( get_the_title( $event->ID ) ); ?>"
-			>
-			</div>
-		</div>
+		<img
+			src="<?php echo esc_attr( $event->thumbnail->full->url ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/day/event/featured-image.php]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: 9f7494cf51bd11c0a3cc8c278230b605</sub>

File: src/views/v2/list/event/featured-image.php [hunk 2]
```diff
@@ -28,14 +28,18 @@
...
-			<div
-				class="tribe-common-c-image__bg"
-				style="background-image: url('<?php echo esc_url( $event->thumbnail->full->url ); ?>');"
-				role="img"
-				aria-label="<?php echo esc_attr( get_the_title( $event->ID ) ); ?>"
-			>
-			</div>
-		</div>
+		<img
+			src="<?php echo esc_attr( $event->thumbnail->full->url ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/list/event/featured-image.php]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: e373733f32e0427ffbe68df36156c669</sub>

File: src/views/v2/month/calendar-body/day/calendar-events/calendar-event/featured-image.php [hunk 3]
```diff
@@ -29,14 +29,18 @@
...
-			<div
-				class="tribe-common-c-image__bg"
-				style="background-image: url('<?php echo esc_url( $event->thumbnail->full->url ); ?>');"
-				role="img"
-				aria-label="<?php echo esc_attr( get_the_title( $event->ID ) ); ?>"
-			>
-			</div>
-		</div>
+		<img
+			src="<?php echo esc_attr( $event->thumbnail->full->url ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day/calendar-events/calendar-event/featured-image.php]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: a062848c00565629cae84c0ee40e3fcf</sub>

File: src/views/v2/month/calendar-body/day/calendar-events/calendar-event/tooltip/featured-image.php [hunk 4]
```diff
@@ -27,14 +27,18 @@
...
-			<div
-				class="tribe-common-c-image__bg"
-				style="background-image: url('<?php echo esc_url( $event->thumbnail->full->url ); ?>');"
-				role="img"
-				aria-label="<?php echo esc_attr( get_the_title( $event->ID ) ); ?>"
-			>
-			</div>
-		</div>
+		<img
+			src="<?php echo esc_attr( $event->thumbnail->full->url ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/calendar-body/day/calendar-events/calendar-event/tooltip/featured-image.php]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: 4f415454083280d4ce9c27ddd41bf014</sub>

File: src/views/v2/month/mobile-events/mobile-day/mobile-event/featured-image.php [hunk 5]
```diff
@@ -22,13 +22,17 @@
...
-		<div
-			class="tribe-common-c-image__bg"
-			style="background-image: url('<?php echo esc_url( $event->thumbnail->full->url ); ?>');"
-			role="img"
-			aria-label="<?php echo esc_attr( get_the_title( $event->ID ) ); ?>"
-		>
-		</div>
-	</div>
+	<img
+		src="<?php echo esc_attr( $event->thumbnail->full->url ); ?>"
```
Comment (tr1b0t) [inline src/views/v2/month/mobile-events/mobile-day/mobile-event/featured-image.php]: :no_entry_sign: Wrong escaping function. href and src attributes should be escaped by `esc_url()`, not by `esc_attr()`.

<sub>hash: d77db59780eaaeef85d6fadf806b846d</sub>

**Comment by paulmskim** [comment]
Tests seem to be failing here.

Also, @juanfra are we adding a note to recommend 16:9 aspect ratio for featured images?

**Comment by juanfra** [comment]
@paulmskim thanks for the CR! Yes, it seems that the error in tests is inherited. I can see it in my other PR: https://github.com/moderntribe/the-events-calendar/pull/2761 - trying to track down the cause to see if I can unblock these.

With regards to the recommendation, you would like to have that in the code? At some point we're gonna add the option to have a "controlled aspect ratio".

**Comment by paulmskim** [comment]
Oh, I was just going off of what the ticket was saying, about adding a note to recommend users to upload 16:9 aspect ratio images.

**Comment by bordoni** [comment]
Once https://github.com/moderntribe/the-events-calendar/pull/2766 is merged the tests will pass. cc: @lucatume

**Comment by juanfra** [comment]
Added a note for aspect ratio recommendation on the "Featured image" box. This is ready for a final pass: https://www.loom.com/share/8d1be14c4fe3429d9e7eca9fcc723f59

File: src/Tribe/Views/V2/Hooks.php [hunk 6]
```diff
@@ -282,9 +283,28 @@ public function filter_excerpt_length( $length ) {
...
+	 *
+	 * @return string The modified html, if required.
+	 */
+	public function filter_admin_post_thumbnail_html( $html ) {
+
+		if ( TEC::POSTTYPE !== get_current_screen()->post_type ) {
+			return $html;
+		}
+
+		return $html . __( 'We recommend a 16:9 aspect ratio for featured images.', 'the-events-calendar' );
```
Comment (paulmskim) [inline src/Tribe/Views/V2/Hooks.php]: only suggestion here is to wrap this in a `<p>` tag with classes `hide-if-no-js howto`

**Pull request merged**